### PR TITLE
Move config to Ion

### LINKED ISSUE #141: Shell config
As we approach the 2.0 release, we need to make sure the new shell config strategy is something we're happy with going forward. I think there are three broad questions:

1. Is the location and name right?
2. Is the format right?
3. Are the options right?

### Location

We're currently using the xdg crate but I'm not very happy with the outcome. Every single time I've tried to customize the config I can't remember the location! Here they are, for each platform:

- OSX: `~/Library/Application Support/qldbshell/default_config.toml`
- Linux: `~/.config/qldbshell/default_config.toml`
- Windows: `~\AppData\Local\qldbshell\default_config.toml`

I would personally prefer that the OSX and Linux paths were the same. There seems to be a split in the ecosystem around this idea. For example, fish shell uses `~/.config` regardless of OSX/Linux.

On the other hand, the crate we're using already supports `XDG_CONFIG_HOME` as an environment variable, so there is no real reason why the user couldn't change this themselves. I'm not sure there is value in us overriding the default here.

What I think it comes down to is it's really hard to know where to place the config file. Part of the problem is the CLI generator we're using only supports static strings in the help text, so if you run `qldb --help` you don't get a hint where to place the file:

```
qldb --help
< snip >
OPTIONS:
    -c, --config <config>
            Config file to load. By default, this file is in $XDG_CONFIG_HOME/qldbshell/default_config.toml

❯ echo $XDG_CONFIG_HOME
< nothing >
```

I think we should try fix that issue at least and just make peace with the XDG defaults on OSX being weird.

### Format

I think the main question here is whether we want TOML or Ion (text). When we started the 2.0 version it seemed like TOML made sense, but now I'm questing whether that was the right choice. Ion is pretty easy to write out by hand, and the reasons TOML has been useful (e.g. comments) can also be gained through Ion. Given QLDB users already need to know about Ion to use the system, it doesn't make sense to me to introduce another format unless there is some material benefit.

### Are the options right?

Is the structure and names what we want? I think so, but I've done 20 iterations on this now so I'm a little dizzy thinking it through. The current pattern is hierarchical which gives both DRY and convention over configuration.

### PR DESCRIPTION
**Pull request by marcbowes**
Fixes #141

### CODE REVIEW

File: src/settings/config.rs [hunk 1]
```diff
@@ -91,7 +181,7 @@ impl ShellConfig {
         let config_dir = dirs::config_dir().ok_or(anyhow!("$XDG_CONFIG_HOME not set"))?;
         let shell_dir = config_dir.join("qldbshell");
         fs::create_dir_all(&shell_dir)?;
-        Ok(shell_dir.join("default_config.toml"))
+        Ok(shell_dir.join("config.toml"))
```
[the remark this answers is not in the document]
Comment (marcbowes) [reply src/settings/config.rs]: yup - you're too fast on the review (it was a WIP). I'm done now, I think, so if you spot any more of these... :)

File: README.md [hunk 2]
```diff
@@ -41,50 +41,62 @@ qldb --ledger <your-ledger>
 
 ### Configuration
 
-After installation, the Shell will load default config file located at `$XDG_CONFIG_HOME/qldbshell/default_config.toml` during initialization. For example on OSX this will typically be at `/Users/username/Library/Application Support/qldbshell/default_config.toml`. You will need to create the `default_config.toml` config file manually after installation. The config file uses [TOML syntax](https://github.com/toml-lang/toml) and the format is as follows:
-
+After installation, the Shell will load default config file located at
+`$XDG_CONFIG_HOME/qldbshell/config.ion` during initialization. For example on
+OSX this will typically be at `~/Library/Application Support/qldbshell/config.ion`. If such a file doesn't exist, the shell will run
```
Comment (brack) [inline README.md:46]: i see that this PR also updates the macos config location in addition to changing to ion -- if that is the case, this should be updated (and maybe should specify our intentional decision to override?)

File: src/settings/config.rs [hunk 3]
```diff
@@ -1,97 +1,301 @@
...
+            );
+        };
+
+        ui.format = if let Some(elem) = value.get("format") {
+            let format = elem
+                .as_str()
+                .ok_or(usage_error("`ui.format` should be a string"))?;
+            FormatMode::from_str(format)?
+        } else {
+            FormatMode::Ion
```
Comment (guyilin-amazon) [inline src/settings/config.rs:152]: Is the `ui.format` already `Ion` after `let mut ui = UiConfig::default();`? Is the `else` logic needed here?

Comment (marcbowes) [reply src/settings/config.rs:152]: That's a good point! I'll update this and all the other places.

**Comment by marcbowes** [comment]
@guyilin-amazon did you test this locally? I did, but want to make sure it had more than just me trying it..

**Comment by guyilin-amazon** [comment]
> @guyilin-amazon did you test this locally? I did, but want to make sure it had more than just me trying it..

I have tested it locally and it also works on my machine

**Pull request merged**